fbw-a32nx/src/systems/instruments/src/NDv2/shared/map/CanvasMap.tsx

```diff
@@ -0,0 +1,414 @@
+            newTraffic.forEach((intruder: NdTraffic) => {
+                const latLong: Coordinates = { lat: intruder.lat, long: intruder.lon };
+                let [x, y] = this.mapParams.coordinatesToXYy(latLong);
+
+                let tcasMask;
+                switch (this.mapMode.get()) {
+                case EfisNdMode.ARC:
+                    tcasMask = [
+                        [-384, -310], [-384, 0], [-264, 0], [-210, 59], [-210, 143],
+                        [210, 143], [210, 0], [267, -61], [384, -61],
+                        [384, -310], [340, -355], [300, -390], [240, -431.5],
+                        [180, -460], [100, -482], [0, -492], [-100, -482],
+                        [-180, -460], [-240, -431.5], [-300, -390], [-340, -355],
+                        [-384, -310],
+                    ];
+                    break;
+                case EfisNdMode.ROSE_NAV:
+                case EfisNdMode.ROSE_ILS:
+                case EfisNdMode.ROSE_VOR:
+                    tcasMask = [
+                        [-340, -227], [-103, -227], [-50, -244],
+                        [0, -250], [50, -244], [103, -227], [340, -227],
+                        [340, 180], [267, 180], [210, 241], [210, 383],
+                        [-210, 383], [-210, 300], [-264, 241], [-340, 241], [-340, -227],
+                    ];
+                    break;
+                default:
+                    break;
+                }
+
+                // Full time option installed: For all ranges except in ZOOM ranges NDRange > 9NM
+                if (tcasMask !== undefined && !MathUtils.pointInPolygon(x, y, tcasMask)) {
+                    if (intruder.intrusionLevel < TaRaIntrusion.TA) {
+                        // Remove if beyond viewable range
+                        return;
+                    }
+                    const ret: [number, number] | null = MathUtils.intersectWithPolygon(x, y, 0, 0, tcasMask);
+                    if (ret) [x, y] = ret;
+                }
+                intruder.posX = x;
+                intruder.posY = y;
+                this.traffic.push(intruder);
+            });
+        }
+        this.trafficLayer.data = this.traffic; // Populate with new traffic
+    }
+
+    private handleFrame(deltaTime: number) {
+        // console.log(`center: lat=${this.props.mapCenterLat.get()}, long=${this.props.mapCenterLong.get()}`);
+
+        const canvas = this.canvasRef.instance;
+        const context = canvas.getContext('2d');
+
+        const size = 768;
+
+        context.clearRect(0, 0, size, size);
+
+        switch (this.mapMode.get()) {
+        case EfisNdMode.ARC:
+            context.clip(ARC_CLIP);
+            break;
+        case EfisNdMode.ROSE_NAV:
+        case EfisNdMode.ROSE_ILS:
+        case EfisNdMode.ROSE_VOR:
+            context.clip(new Path2D('M45,155 L282,155 a250,250 0 0 1 204,0 L723,155 L723,562 L648,562 L591,625 L591,768 L174,768 L174,683 L122,625 L45,625 L45,155'));
```

Comment:
Maybe we can save this in a constant statically instead of instantiating a new `Path2D` every frame? The same with the lines below.